goal_src/engine/camera/cam-start.gc

```diff
@@ -5,3 +5,78 @@
 ;; name in dgo: cam-start
 ;; dgos: GAME, ENGINE
 
+;; definition for function cam-stop
+(defun cam-stop ()
+  (kill-by-name 'camera-master *active-pool*)
+  (kill-by-name 'camera-slave *active-pool*)
+  (kill-by-name 'camera-combiner *active-pool*)
+  (set! *camera* #f)
+  (set! *camera-combiner* #f)
+  (let ((v0-3 cam-string))
+   (set! *camera-base-mode* v0-3)
+   v0-3
+   )
+  )
+
+;; definition for function cam-start
+;; INFO: Return type mismatch int vs none.
+(defun cam-start ((arg0 symbol))
+  (cam-stop)
+  (let ((s5-0 (get-process *camera-dead-pool* camera-combiner #x4000)))
+   (when s5-0
+    (let ((t9-2 (method-of-type camera-combiner activate)))
+     (t9-2
+      (the-as camera-combiner s5-0)
+      *camera-pool*
+      'camera-combiner
+      (the-as pointer #x70004000)
+      )
+     )
+    (let ((t9-3 (the-as (function process function) run-function-in-process))
+          (a0-3 s5-0)
+          )
+     (let ((a1-2 cam-combiner-init))
+      )
+     (t9-3 a0-3)
+     )
+    (let ((v1-3 (-> s5-0 ppointer)))
+     )
+    )
+   )
+  (let* ((s5-1 (get-process *camera-master-dead-pool* camera-master #x4000))
+         (v1-5 (when s5-1
+                (let ((t9-5 (method-of-type camera-master activate)))
+                 (t9-5
+                  (the-as camera-master s5-1)
+                  *camera-pool*
+                  'camera-master
+                  (the-as pointer #x70004000)
+                  )
+                 )
+                (let ((t9-6 (the-as (function process function) set-to-run))
+                      (a0-6 (-> s5-1 main-thread))
+                      )
+                 (let ((a1-5 cam-master-init))
+                  )
+                 (t9-6 (the-as process a0-6))
+                 )
+                (-> s5-1 ppointer)
+                )
+          )
+         )
+   (set! *camera* (the-as camera-master (if v1-5
+                                         (-> v1-5 0 self)
+                                         )
+                   )
+    )
+   )
+  (if arg0
+   (reset-cameras)
+   )
+  (let ((v0-8 0))
+   )
+  (none)
+  )
+
+;; failed to figure out what this is:
+(cam-start #f)
```

Comment:
I suspect this will cause a crash because of the scratchpad thing.  I will work this: https://github.com/water111/jak-project/issues/645

Reply:
This is probably causing the crash in tests, you can comment this out with a TODO for now. It's probably calling some function we don't have yet.